examples/python-guide/dask/binary-classification.py

```diff
@@ -1,10 +1,15 @@
+import sys
+
 import dask.array as da
 from distributed import Client, LocalCluster
 from sklearn.datasets import make_blobs
 
 import lightgbm as lgb
 
 if __name__ == "__main__":
+    if not sys.platform.startswith('linux'):
```

Comment:
If the examples were working on Mac, can we remove this check until something breaks? These are supposed to be examples that teach users how to use `lightgbm.dask`, and I'm really uncomfortable with including details only necessary for CI in them.

Reply:
> including details only necessary for CI in them.

I don't think these details are CI-related only. Imagine, user is using macOS and trying to get familiar with LightGBM's Dask-package. There is no info about that we only support Linux for now anywhere in the docs. So user runs example, gets a crash and feels disappointed (maybe posts a new issue here).

> If the examples were working on Mac

I guess that according to #3782, this is just a happy coincidence.

However, this is not my strong opinion and if you still think that it is better to allow users execute examples that we are know **may** crash on their systems, I'll be happy to revert these `if` guards.

Reply:
I think they are CI-specific because of the use of `sys.exit(0)`. If a user copies that code and runs it in a Python REPL, their Python session will end. If they run it in a Jupyter notebook, they'll get an error like this

> SystemExit: 0
/Users/jlamb/miniconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3425: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)

On this point:

> There is no info about that we only support Linux for now anywhere in the docs.
> I guess that according to #3782, this is just a happy coincidence

I think that's an oversight, and that we should have a note at https://lightgbm.readthedocs.io/en/latest/Parallel-Learning-Guide.html#dask saying that Dask features are only tested on Linux.

While I was developing these examples in #3814 , I tested them on my Mac. So I do actually expect them to continue working on Mac. They're much simpler than the tests in `test_dask.py`, which is why I think we're unable to run `test_dask.py` in Mac CI jobs.

I'm in favor of reverting these `if` guards and instead adding a note to the documentation saying that `lightgbm.dask` is only tested on Linux.

Reply:
> If a user copies that code and runs it in a Python REPL, their Python session will end. If they run it in a Jupyter notebook, they'll get an error like this

It is suggested to run these examples only from command line and from nothing else.

> After installing the package and its dependencies, any of the examples here can be run with a command like this:
    `python binary-classification.py`

> I'm in favor of reverting these if guards and instead adding a note to the documentation saying that lightgbm.dask is only tested on Linux.

Sure, done!